File: 프로그래머스/unrated/181922. 수열과 구간 쿼리 4/수열과 구간 쿼리 4.py
def solution(arr, queries):  
    for query in queries:
        index_arr = [num for num in range(query[0], query[1]+1)]
        for i, num in enumerate(arr):
            if i in index_arr and i%query[2] == 0:
                arr[i] += 1 
    return arr

File: 프로그래머스/unrated/181922. 수열과 구간 쿼리 4/수열과 구간 쿼리 4.java
class Solution {
    public int[] solution(int[] arr, int[][] queries) {
        int start, end, pivot;
        
        for(int[] query:queries) {
            start = query[0];
            end   = query[1];
            pivot = query[2];
            
            for(int num = start; num <= end; num++) {
                if(num%pivot == 0) {
                    arr[num] += 1;
                }
            }
        }
        return arr;
    }
}